pylint/checkers/strings.py

```diff
@@ -787,3 +871,66 @@ def str_eval(token):
     if token[0:3] in ('"""', "'''"):
         return token[3:-3]
     return token[1:-1]
+
+
+def is_long_string(token: str) -> bool:
+    """Is this token a "longstring" (is it well-formed and triple-quoted)?
+
+    Long strings are triple-quoted as defined in
+    https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
+
+    Args:
+        token: The token to be parsed.
+
+    Returns:
+        A boolean representing whether or not this token matches a longstring
+        regex.
+    """
+    single_quoted_regex = "(%s)?'''.*'''\Z" % "|".join(_STRING_AND_BYTES_PREFIXES)
+    double_quoted_regex = '(%s)?""".*"""\Z' % "|".join(_STRING_AND_BYTES_PREFIXES)
+
+    return re.match(single_quoted_regex, token, re.DOTALL) or re.match(
+        double_quoted_regex, token, re.DOTALL
+    )
+
+
+def get_quote_delimiter(token: str) -> Optional[str]:
+    """Returns the quote character used to delimit this token string.
+
+    This function does little checking for whether the token is a well-formed
+    string.
+
+    Args:
+        token: The token to be parsed.
+
+    Returns:
+        A string containing solely the first quote delimiter character in the
+        passed string.  If no quote delimiter characters are present, `None` is
+        returned instead.
+    """
+    match = re.match(
+        "(%s)?(\"|')" % "|".join(_STRING_AND_BYTES_PREFIXES), token, re.DOTALL
+    )
+    if match:
+        return match.group(2)
+
+
+def is_quote_delimiter_chosen_freely(token: str) -> bool:
+    """Was there a non-awkward option for the quote delimiter?
+
+    Args:
+        token: The quoted string whose delimiters are to be checked.
+
+    Returns:
+        Whether there was a choice in this token's quote character that would
+        not have involved backslash-escaping an interior quote character.  Long
+        strings are excepted from this analysis under the assumption that their
+        quote characters are set by policy.
+    """
+    quote_delimiter = get_quote_delimiter(token)
+    non_delimiter = '"' if quote_delimiter == "'" else "'"  # the unchosen option
+    return (
+        quote_delimiter
+        and not is_long_string(token)
+        and not re.search(non_delimiter, str_eval(token))
```

Comment:
you can just say `non_delimiter not in str_eval(token)` here

Reply:
Oh!  That's good to know.   Thanks!